### DISCUSSION TITLE
Pain points in Firebase and how Supabase can makes things easy?

### DISCUSSION THREAD

**Discussion by subhendukundu** [supabase/supabase#565] [Questions] [answered]
So I have been keeping my eye on Supabase for a while now. And I wanted to contribute to a few things.
We all like firebase for a lot of reasons and we hate certain reasons. Definitely, Supabase is taking learning from everywhere. I think it's the right time or place to note down the pain points. And make the community around it to make adaptation easy. A great inspiration would be react-hook-form.com (Not relevant much, but I had seen it grow from 500 weekly downloads to this extend) and how the community is growing there.

If there's already a place we are adding these points, please share the link, I will add my points there. Otherwise, let's start adding here?

## PainPonts / Place to improve

1. Custom domain name with storage (I understand Supabase doesn't have the storage up yet).
2. Flexibility when it comes to choosing endpoints APIs. If I am making a social media app with Supabase I might not need lot of chunk of data everywhere. Graphql is amazing for that (Unless you are waiting for [react server components](https://reactjs.org/blog/2020/12/21/data-fetching-with-react-server-components.html) if you are using react).
3. Execution times in firebase function even for a small logic 50ms to 500ms.
4. Custom authentication could be easier. https://www.back4app.com/ does it to some extent.

**Comment by subhendukundu** [comment]
https://twitter.com/elitasson/status/1350014139079520256/photo/1

Found this image from elitasson's tweet. Seems this talk more about it too.

**Comment by kiwicopple** [comment, marked as answer]
Hi @subhendukundu thanks for dropping in. I'll answer this inline:


> Custom domain name with storage (I understand Supabase doesn't have the storage up yet).

We're working on storage now :). We'll release it at the end of Q1

> Flexibility when it comes to choosing endpoints APIs. If I am making a social media app with Supabase I might not need lot of chunk of data everywhere. Graphql is amazing for that (Unless you are waiting for react server components if you are using react).

I'm not too sure what you're asking here - would you like us to add GraphQL? It's unlikely we will as PostgREST already offers [relationship queries](https://supabase.io/docs/client/select#query-foreign-tables) and, even better, it runs the queries natively on the database. We prefer to build tooling around the database and keep the middleware as thin as possible - this is better for performance.
If you know a technical reason why GraphQL is an advantage, let us know - we're open to it if there is a good reason, but right now we want to keep our surface area small

>   Execution times in firebase function even for a small logic 50ms to 500ms.

Right now you can execute [procedures](https://supabase.io/docs/client/rpc). We'll wrap some tooling around this, and we'll build a more robust offering in Q2.

>   Custom authentication could be easier. https://www.back4app.com/ does it to some extent.

Let us know if you're struggling with the Auth - we're making it as simple as possible, so any suggestions for improvements would be great.
